### PR TITLE
Assignment

### PR DESCRIPTION
**Pull request by ashima-24**

### CODE REVIEW

File: conv.py [hunk 1]
```diff
@@ -1,22 +1,45 @@
 import socket
-s=socket.socket(socket.AF_INET,socket.SOCK_STREAM)
-host='snoopy.mpi-inf.mpg.de'
-port=666
-pos=0
-s.connect((host,port))
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Comment (ashish-17) [inline conv.py:3]: space above this!

File: conv.py [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
+HOST = 'snoopy.mpi-inf.mpg.de'
+PORT = 666
+s.connect((HOST, PORT))
 s.send(b"HELLO.\n")
```
Comment (ashish-17) [inline conv.py:10]: space above this!

File: conv.py [hunk 3]
```diff
@@ -1,22 +1,45 @@
...
-        if not msg:
-                   break
-        else:
-             data+=msg
-             #print(msg)
-#             f.write(msg)
-file=open("conv.txt",'rb+')
-file.seek(56)
-file.write(b"62531")
+data = b""
```
Comment (ashish-17) [inline conv.py]: space above this!

File: conv.py [hunk 4]
```diff
@@ -1,22 +1,45 @@
...
+print(fil)
+b = data.split(b'\n')[3]
+print(b)
+img = b.split()[1]
+print(img) # no of bytes of images
+
+H = data.split(b'BYTES')
+t = H[1].split(b'TOKEN')
+
+img1 = bytearray(t[0])
```
Comment (ashish-17) [inline conv.py]: Why do we need a bytearray?

File: conv.py [hunk 5]
```diff
@@ -1,22 +1,45 @@
...
+while 1:
+       msg = s.recv(1024)
+       if not msg:
+                 break
+       else:
+            data += msg
+
+start = data.split(b'\n')
+helo = data.split(b'\n')[0]
+print(helo)
```
Comment (ashish-17) [inline conv.py]: print statements are for debugging..remove them once they have served the purpose.

File: conv.py [hunk 6]
```diff
@@ -1,22 +1,45 @@
...
-        else:
-             data+=msg
-             #print(msg)
-#             f.write(msg)
-file=open("conv.txt",'rb+')
-file.seek(56)
-file.write(b"62531")
+data = b""
+while 1:
+       msg = s.recv(1024)
```
Comment (ashish-17) [inline conv.py]: 1024 is a constant.
Declare it as a constant at the top and then use that name

File: conv.py [hunk 7]
```diff
@@ -1,22 +1,45 @@
...
+print(img) # no of bytes of images
+
+H = data.split(b'BYTES')
+t = H[1].split(b'TOKEN')
+
+img1 = bytearray(t[0])
+img3 = img1.lstrip(b'\n')
+print(img3)   # raw bytes of image
+
+with open('CAT.jpeg', 'wb') as i:
```
Comment (ashish-17) [inline conv.py]: CAT.jpeg is also a constant

File: conv.py [hunk 8]
```diff
@@ -1,22 +1,45 @@
...
+    i.write(img3)
+
+tok = t[1].split(b'\n')
+token = tok[0].split(b':')
+print(token[1])  # 32 bit token
+
+with open('TOKEN.txt', 'wb') as w:
```
Comment (ashish-17) [inline conv.py]: TOKEN.txt is also a constant

File: conv.py (hunk 4 above)
Comment (ashima-24) [reply conv.py]: byte operation peform k ly lstrip

File: conv.py [hunk 9]
```diff
@@ -1,22 +1,49 @@
...
-        else:
-             data+=msg
-             #print(msg)
-#             f.write(msg)
-file=open("conv.txt",'rb+')
-file.seek(56)
-file.write(b"62531")
+
+data = b""
+size=1024
```
Comment (ashish-17) [inline conv.py]: constants should be in capital

Comment (ashima-24) [reply conv.py]: ok file name bhi ?

File: conv.py [hunk 10]
```diff
@@ -1,22 +1,46 @@
...
+
+DATA = b""
+SIZE=1024
+CAT='CAT.jpeg'
+TO='TOKEN.txt'
+
+while True:
+       msg = s.recv(SIZE)
+       if not msg:
+                 break
```
Comment (ashish-17) [inline conv.py]: is this working?
break se pehle extra tab lagri hai!!!

File: conv.py [hunk 11]
```diff
@@ -1,22 +1,46 @@
 import socket
-s=socket.socket(socket.AF_INET,socket.SOCK_STREAM)
-host='snoopy.mpi-inf.mpg.de'
-port=666
-pos=0
-s.connect((host,port))
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Comment (ashish-17) [inline conv.py:3]: is line se pehle space!!

File: conv.py (hunk 10 above)
Comment (ashima-24) [reply conv.py]: nhi chlra hai yeh h

**Review by ashish-17 — APPROVED** [review]

**Review by ashish-17 — APPROVED** [review]

**Pull request merged**